src/main/java/page/Parser.java

```diff
@@ -0,0 +1,95 @@
+package page;
+
+import page.quest.*;
+
+public class Parser {
+
+    public boolean parseExecute(String input, Ui ui, Storage storage, QuestLog questLog) throws PageException {
+        switch (input) {
+            case "bye":
+                storage.saveData(questLog);
+                ui.printBye();
+                return true;
+            case "log":
+                ui.printQuestLog(questLog);
+                break;
+            case "help":
+                ui.printHelpMessage();
+                break;
+            default:
+                String[] splitInput = input.split(" ", 2);
+                switch (splitInput[0]) {
+                    case "todo":
+                        if (splitInput.length == 1) {
+                            throw new PageException("Sorry, please include a task description!");
+                        } else {
+                            Todo newTodo = questLog.addTodo(splitInput[1]);
+                            ui.printQuestAdded(newTodo);
+                        }
+                        break;
+                    case "deadline":
+                        if (splitInput.length == 1) {
+                            throw new PageException("Sorry, please include a task description!");
+                        } else {
+                            String[] splitBy = splitInput[1].split(" /by ");
+                            if (splitBy.length == 1) {
+                                throw new PageException("Sorry, please include a /by time after the task description!");
+                            } else {
+                                Deadline newDeadline = questLog.addDeadline(splitBy[0], splitBy[1]);
+                                ui.printQuestAdded(newDeadline);
+                            }
+                        }
+                        break;
+                    case "event":
+                        if (splitInput.length == 1) {
+                            throw new PageException("Sorry, please include a task description!");
+                        } else {
+                            String[] splitFrom = splitInput[1].split(" /from ");
+                            if (splitFrom.length == 1) {
+                                throw new PageException("Sorry, please include a /from time!");
+                            } else {
+                                String[] splitTo = splitFrom[1].split(" /to ");
+                                if (splitTo.length == 1) {
```

Comment:
Very deep nesting here, a bit hard to read, maybe can extract out certain code as a different method